internal/pkg/generator/forwarder/generator.go

```diff
@@ -0,0 +1,61 @@
+package forwarder
+
+import (
+	"fmt"
+	"os"
+
+	yaml "gopkg.in/yaml.v2"
+
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/pkg/generators/forwarding"
+	"github.com/openshift/cluster-logging-operator/pkg/k8shandler"
+	"github.com/openshift/cluster-logging-operator/pkg/logger"
+)
+
+const (
+	//these are fixed at the moment
+	logCollectorType         = logging.LogCollectionTypeFluentd
+	includeLegacyForward     = false
+	includeLegacySyslog      = false
+	useOldRemoteSyslogPlugin = false
+)
+
+func Generate(clfYaml string, includeDefaultLogStore bool) (string, error) {
+
+	generator, err := forwarding.NewConfigGenerator(
+		logCollectorType,
+		includeLegacyForward,
+		includeLegacySyslog,
+		useOldRemoteSyslogPlugin)
+	if err != nil {
+		return "", fmt.Errorf("Unable to create collector config generator: %v", err)
+	}
+
+	forwarder := &logging.ClusterLogForwarder{}
+	err = yaml.Unmarshal([]byte(clfYaml), forwarder)
+	if err != nil {
+		logger.Errorf("Error Unmarshalling %q: %v", clfYaml, err)
+		os.Exit(1)
+	}
+	logger.DebugObject("Unmarshalled %s", forwarder)
```

Comment:
@blockloop you did the logging cleanup on EO, is there a story up for doing the same to CLO? This issue should be fixed at the same time - get rid of DebugObject and replace it with the generic log statetment and test.YAMLString(forwarder) to write the object.

Reply:
> @blockloop you did the logging cleanup on EO, is there a story up for doing the same to CLO? This issue should be fixed at the same time - get rid of DebugObject and replace it with the generic log statetment and test.YAMLString(forwarder) to write the object.

@alanconway there is a card this spring that is assigned to @vparfonov to work with @blockloop

Reply:
> How to see output from this function? i have never seen any output from this function in practice.
> Since this is a test, perhaps we change this function to generate outpout without enabling debug logs

you have to modify the loglevel like LOG_LEVEL=debug.  and it seems it only appears on test failure.  Also, not sure if the DEBUG log works correctly for cases like "trace"